src/components/PopoverMenu/BasePopoverMenu.js

```diff
@@ -26,13 +26,88 @@ const defaultProps = {
 };
 
 class BasePopoverMenu extends PureComponent {
+    constructor(props) {
+        super(props);
+        this.state = {
+            activeMenuIndex: -1,
+        };
+        this.onModalHide = this.onModalHide.bind(this);
+    }
+
+    componentDidMount() {
+        if (!this.props.allowKeyboardNavigation) {
+            return;
+        }
+        this.setupEventHandlers();
+    }
+
+    componentWillUnmount() {
+        if (!this.props.allowKeyboardNavigation) {
+            return;
+        }
+        this.cleanupEventHandlers();
+    }
+
+    onModalHide() {
+        this.setState({activeMenuIndex: -1});
+        this.props.onMenuHide();
+    }
+
+    setupEventHandlers() {
+        if (!document) {
+            return;
+        }
+        this.keyDownHandler = (keyBoardEvent) => {
+            if (keyBoardEvent.key.startsWith('Arrow')) {
+                this.highlightActiveMenu(keyBoardEvent.key);
+                return;
+            }
+
+            if (keyBoardEvent.key === 'Enter' && this.state.activeMenuIndex !== -1) {
+                this.props.onItemSelected(this.props.menuItems[this.state.activeMenuIndex]);
+            }
+        };
+        document.addEventListener('keydown', this.keyDownHandler, true);
+    }
+
+    highlightActiveMenu(arrowKey) {
```

Comment:
Missing JSDoc

Reply:
Keyboard shortcuts are more to do with hotkeys. This was more of arrow navigation, hence used this